great_expectations/datasource/data_connector/util.py

```diff
@@ -249,7 +249,8 @@ def _invert_regex_to_data_reference_template(
 
     # print("-"*80)
     parsed_sre = sre_parse.parse(str(regex_pattern))
-    for token, value in parsed_sre:  # type: ignore[attr-defined]
+    for thing, char in zip(parsed_sre, list(str(regex_pattern))):  # type: ignore[attr-defined]
+        token, value = thing
```

Comment:
Couldn't think of a great name but how about this?
```suggestion
    for parsed_sre_tuple, char in zip(parsed_sre, list(str(regex_pattern))):  # type: ignore[attr-defined]
        token, value = parsed_sre_tuple
```